Content.Shared/Teleportation/Systems/LinkedEntitySystem.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System.Linq;
+using Content.Shared.Teleportation.Components;
+using Robust.Shared.GameStates;
+
+namespace Content.Shared.Teleportation.Systems;
+
+/// <summary>
+///     Handles symmetrically linking two entities together, and removing links properly.
+/// </summary>
+public sealed class LinkedEntitySystem : EntitySystem
+{
+    [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<LinkedEntityComponent, ComponentShutdown>(OnLinkShutdown);
+
+        SubscribeLocalEvent<LinkedEntityComponent, ComponentGetState>(OnGetState);
+        SubscribeLocalEvent<LinkedEntityComponent, ComponentHandleState>(OnHandleState);
+    }
+
+    private void OnGetState(EntityUid uid, LinkedEntityComponent component, ref ComponentGetState args)
+    {
+        args.State = new LinkedEntityComponentState(component.LinkedEntities);
+    }
+
+    private void OnHandleState(EntityUid uid, LinkedEntityComponent component, ref ComponentHandleState args)
+    {
+        if (args.Current is LinkedEntityComponentState state)
+            component.LinkedEntities = state.LinkedEntities;
+    }
+
+    private void OnLinkShutdown(EntityUid uid, LinkedEntityComponent component, ComponentShutdown args)
+    {
+        // Remove any links to this entity when deleted.
+        foreach (var ent in component.LinkedEntities.ToArray())
+        {
+            if (!Deleted(ent) && !Terminating(ent) && TryComp<LinkedEntityComponent>(ent, out var link))
+            {
```

Comment:
Should probably check lifestage instead? Deleted and terminating will do separate metadata resolves.

Reply:
tru